### PR TITLE
Remove unused args

### LINKED ISSUE #1907: [Maintenance] Delete unused variables
We currently use `@typescript-eslint/no-unused-vars` eslint rule which is enabled for most things. However, it is disabled for `args`.

The overall goal here would be to re-enable `@typescript-eslint/no-unused-vars` for `args`. Thus disallowing unuser arguments.

https://github.com/actualbudget/actual/blob/08c80b6f585eaa3237100972f0edc06e45e444d1/.eslintrc.js#L54

**Please do not do this as one-big-bang migration.**

### PR DESCRIPTION
**Pull request by jaas666**
Should close #1907 

No linting errors or warnings after changes but I'd still review carefully as there are a bunch of files with changes.

Also, there are two errors that I noticed when running tests but they don't seem to be related to changes on this PR. These errors might need new issues created.

Running `npm run test`:
```
[loot-core]: FAIL src/shared/util.test.ts
[loot-core]:   ● utility functions › number formatting works with space-comma format
[loot-core]: 
[loot-core]:     expect(received).toBe(expected) // Object.is equality
[loot-core]: 
[loot-core]:     Expected: "1 234,56"
[loot-core]:     Received: "1,234.56"
[loot-core]: 
[loot-core]:       69 |     let formatter = getNumberFormat().formatter;
[loot-core]:       70 |     // grouping separator space char is a non-breaking space, or UTF-16 \xa0
[loot-core]:     > 71 |     expect(formatter.format(Number('1234.56'))).toBe('1\xa0234,56');
[loot-core]:          |                                                 ^
[loot-core]:       72 |
[loot-core]:       73 |     setNumberFormat({ format: 'space-comma', hideFraction: true });
[loot-core]:       74 |     formatter = getNumberFormat().formatter;
[loot-core]: 
[loot-core]:       at Object.<anonymous> (src/shared/util.test.ts:71:49)
```

Running `npm run typecheck`:
```
packages/desktop-client/src/components/table.tsx:889:14 - error TS2322: Type 'ForwardRefExoticComponent<TableProps<TableItem> & RefAttributes<TableHandleRef>>' is not assignable to type '<T extends TableItem>(props: TableProps<T> & { ref?: Ref<TableHandleRef>; }) => ReactElement<any, string | JSXElementConstructor<any>>'.
  Type 'ReactNode' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'.
    Type 'string' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'.

889 export const Table: <T extends TableItem>(
                 ~~~~~


Found 1 error in packages/desktop-client/src/components/table.tsx:889
```

### CODE REVIEW

File: packages/desktop-client/src/browser-preload.browser.js [hunk 1]
```diff
@@ -121,9 +121,9 @@ global.Actual = {
   openURLInBrowser: url => {
     window.open(url, '_blank');
   },
-  onEventFromMain: (type, handler) => {},
+  onEventFromMain: () => {},
   applyAppUpdate: () => {},
-  updateAppMenu: isBudgetOpen => {},
+  updateAppMenu: () => {},
```
Comment (Shazib) [inline packages/desktop-client/src/browser-preload.browser.js:126]: Several of these preload functions match types used in the electron preload, so these parameters whilst not used here are used in the electron implementation of these functions.
Personally I'd he hesitant to change any of these as its quite convenient that they have the same interface in both worlds.

**Review by Shazib — COMMENTED** [review]
I would almost certainly break this PR into lots of smaller ones. E.g. all the graphs, then all the mobile pages etc for example.

File: packages/desktop-client/src/browser-preload.browser.js (hunk 1 above)
Comment (jaas666) [reply packages/desktop-client/src/browser-preload.browser.js:126]: What do you suggest? Ignoring the rule for those cases maybe?
Feel free to make any edits you consider necessary to the PR.

**Comment by jaas666** [comment]
> I would almost certainly break this PR into lots of smaller ones. E.g. all the graphs, then all the mobile pages etc for example.

🤔🤔 Do you know of a quick way to split them from this PR? If there's such a thing...

At least all checks passed.

**Comment by Shazib** [comment]
> > I would almost certainly break this PR into lots of smaller ones. E.g. all the graphs, then all the mobile pages etc for example.
> 
> 🤔🤔 Do you know of a quick way to split them from this PR? If there's such a thing...
> 
> At least all checks passed.

Hmm. I think you'll have to branch off this PR and then reset to master (excluding certain files).

As long as this PR is kept synced with master you can keep branching off it and repeating.

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 2]
```diff
@@ -185,29 +185,29 @@ function getInitialMappings(transactions) {
   }
 
   let dateField = key(
-    fields.find(([name, value]) => name.toLowerCase().includes('date')) ||
-      fields.find(([name, value]) => value.match(/^\d+[-/]\d+[-/]\d+$/)),
+    fields.find(([name]) => name.toLowerCase().includes('date')) ||
+      fields.find(([value]) => value.match(/^\d+[-/]\d+[-/]\d+$/)),
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:189]:
```suggestion
      fields.find(([, value]) => value.match(/^\d+[-/]\d+[-/]\d+$/)),
```

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 3]
```diff
@@ -185,29 +185,29 @@ function getInitialMappings(transactions) {
...
   );
 
   let amountField = key(
-    fields.find(([name, value]) => name.toLowerCase().includes('amount')) ||
-      fields.find(([name, value]) => value.match(/^-?[.,\d]+$/)),
+    fields.find(([name]) => name.toLowerCase().includes('amount')) ||
+      fields.find(([value]) => value.match(/^-?[.,\d]+$/)),
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:194]:
```suggestion
      fields.find(([, value]) => value.match(/^-?[.,\d]+$/)),
```

File: packages/desktop-client/src/components/payees/ManagePayeesWithData.js [hunk 4]
```diff
@@ -63,7 +63,7 @@ export default function ManagePayeesWithData({ initialSelectedIds }) {
     };
   }, []);
 
-  async function onUndo({ tables, messages, meta, url }, scroll = false) {
+  async function onUndo({ tables, messages, meta }, _scroll = false) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/payees/ManagePayeesWithData.js:66]: **💬 suggestion:** ‏can just remove it if it's not used

```suggestion
  async function onUndo({ tables, messages, meta }) {
```

File: packages/desktop-client/src/components/sidebar/Accounts.tsx [hunk 5]
```diff
@@ -77,7 +77,7 @@ function Accounts({
     setIsDragging(drag.state === 'start');
   }
 
-  let makeDropPadding = (i, length) => {
+  let makeDropPadding = (i, _length) => {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/Accounts.tsx:80]:
```suggestion
  let makeDropPadding = (i) => {
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Thanks!

A few small bugs and nitpicks, but other than that LGTM.

File: packages/desktop-client/src/components/sidebar/Accounts.tsx (hunk 5 above)
Comment (jaas666) [reply packages/desktop-client/src/components/sidebar/Accounts.tsx:80]: There are some cases like this one. The function definition includes the `length` parameter which isn't used in the function so my first thought was to remove it althogether.  But we are calling the function in some places an passing that argument like line [145](https://github.com/actualbudget/actual/blob/28e129c7faf79a597d20c61bfe4f1e37f48e62b6/packages/desktop-client/src/components/sidebar/Accounts.tsx#L145) for example.

Which one do you preffer? Removing the parameter from the function definition and fix the call? or leave the parameter prefixed with an `_` so it gets ignored by the rule?

File: packages/desktop-client/src/components/modals/ImportTransactions.js (hunk 2 above)
Comment (jaas666) [reply packages/desktop-client/src/components/modals/ImportTransactions.js:189]: Can you help me understand this suggestion?

File: packages/desktop-client/src/components/sidebar/Accounts.tsx (hunk 5 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/sidebar/Accounts.tsx:80]: If it's not used - we can safely remove it. :)

**Comment by jaas666** [comment]
I'm out of state until early Dec.
Anyone feel free to modify the PR or submit their own.

If the issue is still open when I get back I'll work on it. Don't want to hold anyone back.

**Pull request closed without merging**